mne_bids/path.py

```diff
@@ -213,11 +214,15 @@ def get_bids_fname(self, kind=None, bids_root=None, extension=None):
 
         return bids_fname
 
-    def update(self, **entities):
+    def update(self, check=False, **entities):
```

Comment:
`check` should default to `True`, to be in sync with `BIDSPath.__init__()` and the docstring.

Reply:
counterpoint here is that you end up having to pass in `check=False` for every update you do if you start with a BIDSPath that isn't compliant (e.g. emptyroom, or derivatives type file). 

Should the `check` in update() default to None, and use the `__init__()` passed in?

Reply:
See the latest commit for how this would work.

Reply:
Personally I don't like the idea of having a "permanent" `.check` attribute. I think one should be forced to explicitly disable the check every time one wishes to do so.

I do agree this is -- currently -- a slight inconvenience when handling derivatives; but it will also serve as a reminder that we need to push for a standardization of those files ;)

Would like to hear the opinions of @agramfort, @jasmainak, and @sappelhoff on this one, too

Reply:
I'm okay with inconvenience then for now. :)